packages/react-kit/src/components/cta/offer/CreateOfferButton.tsx

```diff
@@ -50,7 +50,9 @@ export const CreateOfferButton = ({
       try {
         setIsLoading(true);
         onPendingSignature?.();
-        const isMetaTx = Boolean(coreSdk.isMetaTxConfigSet && signerAddress);
+        const isMetaTx = Boolean(
+          coreSdk.isMetaTxConfigSet && signerAddress && !signerContract
```

Comment:
I feel we are repeating this logic everywhere, could we do something about this?

Reply:
Interesting idea.
However, not sure how to handle that:
- coreSdk.isMetaTxConfigSet is a synchnronous check based on configuration
- while signerAddress and signerContract are async and depends on user actions (Wallet connection)

Reply:
let's add a TODO so we dont forget about this improvement and we dont block this PR

Reply:
done